### PR TITLE
Refactor integration test (step 7) - pipeline updates (linux part)

### PR DESCRIPTION
**Pull request by ultmaster**
Update pipeline YAMLs to use latest utilities and agent pools.

I will NOT take care of frameworkcontroller, kubeflow, openpai, tensorrt.

Update docker start because we need to run docker under sudo now. I don't know how to add a possibly-non-existing user to the docker group at the VM building stage.

Schedules are not enabled yet.

### CODE REVIEW

**Review by J-shang — APPROVED** [review]

File: pipelines/integration-test-hybrid.yml [hunk 1]
```diff
@@ -7,77 +7,56 @@ schedules:
...
-      targetFolder: /tmp/nnitest/$(Build.BuildId)/test
-      overwrite: true
-    displayName: Copy test scripts to remote machine
-    timeoutInMinutes: 10
+  # These parts are exactly same as remote.
+  # Refactor these as templates when the refactor is done.
+  - script: |
+      set -e
+      mkdir -p /tmp/nnitest/$(Build.BuildId)
+      cp -r Dockerfile /tmp/nnitest/$(Build.BuildId)
```
Comment (cruiseliu) [inline pipelines/integration-test-hybrid.yml:35]: `-r`?

Comment (cruiseliu) [inline pipelines/integration-test-hybrid.yml:35]: `-r`?

File: pipelines/integration-test-hybrid.yml [hunk 2]
```diff
@@ -7,77 +7,56 @@ schedules:
...
-      contents: test/vso_tools/interim_patch.py
-      targetFolder: /tmp/nnitest/$(Build.BuildId)
-      overwrite: true
-    displayName: Copy torch patch to remote machine
-    timeoutInMinutes: 10
+  - script: |
+      set -e
+      export worker_ip=localhost
+      export manager_ip=host.docker.internal
+      export password_in_docker=123456
```
Comment (cruiseliu) [inline pipelines/integration-test-hybrid.yml:45]: idk, is it save to publish password in docker?

File: test/vso_tools/start_docker.py [hunk 3]
```diff
@@ -20,14 +21,26 @@
...
+
+    args = parser.parse_args()
+    docker = 'sudo docker' if args.sudo else 'docker'
+    version, container, password = args.version, args.container, args.password
+    uid, gid = os.getuid(), os.getgid()
+
+    run_command(f'{docker} build --build-arg NNI_RELEASE={version} -t nnidev/nni-nightly .')
+    run_command(f'{docker} run --privileged -d -t -p {port}:22 --add-host=host.docker.internal:host-gateway --name {container} nnidev/nni-nightly')
+    run_command(f'{docker} exec {container} groupadd -g {gid} nni')
+    run_command(f'{docker} exec {container} useradd --create-home --password {password} -u {uid} -g {gid} nni')
```
Comment (cruiseliu) [inline test/vso_tools/start_docker.py:39]: What's this for?

**Review by liuzhe-lz — APPROVED** [review]

File: pipelines/integration-test-hybrid.yml (hunk 1 above)
Comment (matluster) [reply pipelines/integration-test-hybrid.yml:35]: I think these are cut-paste.

**Pull request merged**